### PR TITLE
docs: documentation for extension configuration in package.json

### PR DESCRIPTION
**Pull request by ryan-blunden**

### CODE REVIEW

File: doc/extensions/authoring/manifest.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+Name | Required | Type | Details
+---- |:--------:| ---- | -------
+`name` | ✔️ | `string` | Extension identifier: all lowercase, alphanumeric with hyphens and underscores.
+`title` | ✔️ | `string`| The name displayed in the extension registry. Can be used to indicate a [work-in-progress extension](publishing.md#wip-extensions).
+`description` | ✔️ | `string` | A short description of what your extension is and does.
+`version` | | `string` | [Semantic versioning](https://semver.org/) format.
+`publisher` | ✔️ | `string` | Your [Sourcegraph username](development_environment#sourcegraph-com-account-and-the-sourcegraph-cli)
+`license` | | `string` | The type of license chosen.
+`main` | | `string` | Path to the transpiled JavaScript file for your extension.
+`contributes` | | `object` | An object describing the contributions (extension points) for this extension (e.g. menus, buttons, configuration and more).
```
Comment (ryan-blunden) [inline doc/extensions/authoring/manifest.md]: This will link to the contributions page once it is completed.

File: .vscode/launch.json [hunk 2]
```diff
@@ -163,6 +163,18 @@
       "program": "${workspaceRoot}",
       "env": {},
       "args": []
+    },
```
Comment (sqs) [inline .vscode/launch.json]: Bad merge? There are a lot of unrelated changes in here

Comment (ryan-blunden) [reply .vscode/launch.json]: New PR at https://github.com/sourcegraph/sourcegraph/pull/1158

**Comment by ryan-blunden** [comment]
New PR at https://github.com/sourcegraph/sourcegraph/pull/1158
